src/Utilities/TupleSlice.hpp

```diff
@@ -28,17 +35,41 @@ auto tuple_slice(const std::tuple<AllElements...>& tuple) noexcept {
       tuple, std::make_index_sequence<Stop - Start>{});
 }
 
+template <size_t Start, size_t Stop, typename... AllElements>
+auto tuple_slice(std::tuple<AllElements...>&& tuple) noexcept {
+  return tuple_impl_detail::slice_impl<Start>(
+      std::forward<std::tuple<AllElements...>>(tuple),
```

Comment:
you can move the tuple here and below too since you know you own it. `std::forward` is if you have forwarding references, not rvalue references